### ISSUE TITLE
Data Export

### ISSUE THREAD

**Issue by katyedwards** [healthlocker/oxleas-adhd#220]
Oxleas are interested in thinking about how they could export data from the site, either in the case they wanted to close it down or (more likely) as users leave the service.

They want to discuss this with SLaM and maybe work on/ fund something together.

Can you provide high level initial ideas on how this might work and a rough time estimate (range) to inform their discussions.

Thank you!

**Comment by iteles** [comment]
@katbow I think the range here will be from a 'simple' CSV export (we would need to think through how to structure all of the historical data points) of the data to an API.

@katyedwards Usually data exports are not very 'people friendly' in that they are a data dump that you really have to wade through the understand, but with good documentation, can be pulled into other comparable services for example. I just wanted to be sure that was clear!

**Comment by katyedwards** [comment]
@iteles yep that's fine, the intended use as far as I am aware is for the care team so they keep a record in their patient system not for the end user.

**Comment by katbow** [comment]
@katyedwards I'm going to do a quick spike for this before I estimate. Should be about 1 hour following [this](https://medium.com/@SergeyChechaev/elixir-phoenix-everyday-tips-create-xlsx-66cd70268eb4).

**Comment by katbow** [comment]
actually, might not follow that guide -> checking out the differences between 2 packages. [elixlsx](https://github.com/xou/elixlsx) vs [csv](https://github.com/beatrichartz/csv)

**Comment by katyedwards** [comment]
@katbow we don't need a v precise estimate - this is more an initial high level quote so Oxleas have some info to decide what they want to include.

I'd rather you didn't spend an hour on it at this stage as it may not end up in the sprint at all.

Can you give a rough guide, e.g. 1-2 days or 2-4 hours?

**Comment by katbow** [comment]
@katyedwards okay, in that case, from looking at [this guide](https://medium.com/@SergeyChechaev/elixir-phoenix-everyday-tips-create-xlsx-66cd70268eb4) which uses [this library](https://github.com/xou/elixlsx), and also [this guide](http://matthewlehner.net/exporting-csvs-with-phoenix-elixir/) which uses the [csv llibrary](https://github.com/beatrichartz/csv), I think either one will be about 1 day as we will need to go through each table that needs exporting.
